sympy/sets/sets.py

```diff
@@ -153,6 +153,181 @@ def isdisjoint(self, other):
         """
         return self.is_disjoint(other)
 
+    def _union_simplify(self, other):
+        """
+        Try to reduce number of imageset in the args. Helper method for
+        ImageSet `_union `. It helps to get simpler imageset or
+        union of imageset.
+
+        First extract the expression of imageset and store in separate list
+        (self_expr for self and final_list for other). If there is any expr in
+        self_list have difference of pi or -pi with any final_expr expr. then
+        club them into one expr( (2*n + 1)*pi and 2*n*pi  => n*pi). remove
+        previous expr from final_expr and self_expr. Add this new expr in
+        final_expr list. At the end final_list contains the minimum number of
+        expr that can generate all the expressions of `self` and  `other`.
+
+        * Note: This should be used when there is linear expr in ImageSet(s).
+
+        Parameters
+        ==========
+
+        self : S.EmptySet or imageset(s).
+        other : S.EmptySet or  imageset(s)
+.
+        Returns
+        =======
+        Simplified imageset(s) if possible otherwise `self` union `other`.
+
+        Examples
+        ========
+
+        >>> from sympy import Lambda
+        >>> from sympy.sets.fancysets import ImageSet
+        >>> from sympy import pprint
+        >>> from sympy.core import Dummy, pi, S, Symbol
+        >>> n = Dummy('n')
+        >>> pprint(\
+            ImageSet(Lambda(n, 2*n*pi + 3*pi/4), S.Integers)._union_simplify(\
+                ImageSet(Lambda(n, 2*n*pi + 7*pi/4), S.Integers)),\
+                use_unicode = False)
+                3*pi
+        {n*pi + ---- | n in Integers()}
+                 4
+
+
+        >>> pprint((ImageSet(Lambda(n, 2*n*pi + pi/3), \
+            S.Integers) + ImageSet(Lambda(n, 2*n*pi + pi), S.Integers)).\
+            _union_simplify(ImageSet(Lambda(n, 2*n*pi), S.Integers)),\
+            use_unicode = False)
+                                             pi
+        {n*pi | n in Integers()} U {2*n*pi + -- | n in Integers()}
+                                             3
+
+        """
+
+        from sympy.sets.fancysets import ImageSet
+        from sympy.core.function import Lambda
+        from sympy.core import pi
+        from sympy.polys import Poly
+
+        # don't try to reduce imageset(s) from the same union.
```

Comment:
I explained it from next line
```
        # `_solve_trig` function defined in `solveset.py`
        # solves the factors of trig eq and union of these soln is final ans,
        # soln of one factor (say f1) should be clubbed with soln of another
        # factor(say f2) and so on. ImageSets of f1 or f2 should not be
        # reduced(clubbed) with it's own args.
```
My intention is to tell why it returns `self` if  `other` is EmptySet and vice versa.
I will remove first line, it is confusing.
Thanks.